### PR TITLE
Remove neps argument 'searcher_path' and yaml argument searcher_kwargs + New loading design for optimizer

### LINKED ISSUE #69: Enable argument modification for custom optimizer class that has been loaded via yaml
Custom optimizers are specified through neps.run() by using the searcher argument, which inherits from BaseOptimizer. For loading the custom optimizer via 'run_args', we have chosen a design where users do not define and load an instance directly. Instead, they provide the class name and the path to its implementation. The problem arises with the inability to load custom settings for these optimizers. 

Proposed Solution: Adopt the use of 'searcher_kwargs' for custom optimizers, similar to its application for built-in searchers. 

Example:
```
searcher:
   name: BayesianOptimizer
   path: /path/to/optimizer
   initial_design_size: 30
```

### LINKED ISSUE #70: Standardize yaml path definitions for the neps arguments
Currently, neps.run() loads an optimizer with custom settings through two arguments: searcher_path and searcher. The current implementation uses searcher to specify the name of a YAML file within a folder (without including the .yaml extension) and searcher_path for the path to that folder. 
This approach does not align with the YAML usage for other arguments like pipeline_space and run_args, which link directly to a YAML file.

Possible Solutions:
```
searcher: "/path/to/searcher_config.yaml"
```
or we remove this usage completely, because the same functionality of 'searcher_config.yaml' exist already in 'run_args'

### PR DESCRIPTION
**Pull request by danrgll**
- remove neps argument searcher_path
- remove neps yaml argument searcher_kwargs
- new loading design for optimizer (searcher="path/to/my/custom_searcher.yaml")
- rename key 'algorithm' to 'strategy' for searcher
- Defining custom_searcher.yaml within run_args is now possible e.g.
    ```yaml
    searcher: 
       name: "my_searcher" #optional 
       strategy: bayesian_optimization
       initial_design_size: 7
- Closes #70
- Closes #69

### CODE REVIEW

**Review by eddiebergman — CHANGES_REQUESTED** [review]
Looks good as a far as I can see (hard to validate strings everywhere tbh)

One bug that needs fixing, rest is fine!

**Comment by eddiebergman** [comment]
Seems like the test failures here were fixed by @karibbov in #ade69db7c23673fcbe9d2ac2b6dc348810b45bd0 , many thanks :)

File: docs/doc_yamls/full_configuration_template.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ pipeline_space:
   learning_rate:
     lower: 1e-5
     upper: 1e-1
-    log: True  # Log scale for learning rate
+    log: True
```
[the remark this answers is not in the document]
Comment (danrgll) [reply docs/doc_yamls/full_configuration_template.yaml:10]: yeah i think both are interpreted as boolean, but we can definitely align with the more common usage

File: tests/test_neps_api/testing_scripts/user_yaml_neps.py [hunk 2]
```diff
@@ -19,13 +19,12 @@ def run_pipeline(val1, val2):
 # Testing using created yaml with api
 script_directory = os.path.dirname(os.path.abspath(__file__))
 parent_directory = os.path.join(script_directory, os.pardir)
-searcher_path = os.path.join(parent_directory, "testing_yaml")
+searcher_path = os.path.join(parent_directory, "testing_yaml/optimizer_test")
```
[the remark this answers is not in the document]
Comment (danrgll) [reply tests/test_neps_api/testing_scripts/user_yaml_neps.py]: fixed [9406896](https://github.com/automl/neps/pull/105/commits/94068963440f83f7a39327ec67cc929fbcd68f66)
